src/components/dialogs/AddStaffNotationDialog.vue

```diff
@@ -79,233 +115,286 @@ import Vue from 'vue'
+  @Prop({ required: true }) readonly dialog!: boolean
 
   /** Prop to provide attachment selector. */
   @Prop({ default: '' }) readonly attach!: string
 
   /** Prop for the item's display name. */
-  @Prop() readonly displayName!: string
+  @Prop({ required: true }) readonly displayName!: string
 
   /** Prop for the item's name (filing type). */
-  @Prop() readonly name!: FilingTypes
+  @Prop({ required: true }) readonly name!: FilingTypes
 
   /** Prop for the item's dissolution type. */
-  @Prop() readonly dissolutionType!: DissolutionTypes
+  @Prop({ default: null }) readonly dissolutionType!: DissolutionTypes
 
   /** Prop to require court order number regardless the plan of arrangement. */
   @Prop({ default: false }) readonly courtOrderNumberRequired!: boolean
 
-  @Getter getIdentifier!: string
+  // Global getters
   @Getter getCurrentDate!: string
-  @Getter getEntityName!: string
   @Getter getBusinessNumber!: string
-  @Getter getEntityType!: string
   @Getter getEntityFoundingDate!: Date
-
-  /** The notation text. */
-  private notation = ''
-
-  /** Court Order Number */
-  private courtOrderNumber = null as string
-
-  /** Whether filing has plan of arrangement. */
-  private planOfArrangement = false
-
-  /** Whether this component is currently saving. */
-  private saving = false
-
-  /** Court Order component key */
-  private courtOrderKey = 0
-
-  /** Notation form validity */
-  private notationFormValid = false
-
-  /** Notation max length */
-  private notationMaxLength = 2000
-
-  /** Flag to enable validation in this component */
-  private enableValidation = false
-
-  /** Whether this filing is an administrative dissolution */
-  get administrativeDissolution (): boolean {
+  @Getter getEntityName!: string
+  @Getter getEntityType!: string
+  @Getter getIdentifier!: string
+  @Getter getUserKeycloakGuid!: string
+
+  // Properties
+  protected notation = '' // notation text
+  protected courtOrderNumber = '' // court order number
+  protected planOfArrangement = false // whether filing has plan of arrangement
+  protected saving = false // whether this component is currently saving
+  protected courtOrderPoaKey = 0 // court order component key, to force re-render
+  protected enableValidation = false // flag to enable validation
+  protected file: File = null // court order file object
+  protected fileKey: string = null // court order file key
+
+  /** Whether this filing is an Administrative Dissolution. */
+  get isAdministrativeDissolution (): boolean {
     return this.isTypeAdministrativeDissolution({ name: this.name, dissolutionType: this.dissolutionType })
   }
 
-  /** Whether this filing is a put back on */
-  get putBackOn (): boolean {
+  /** Whether this filing is a Put Back On. */
+  get isPutBackOn (): boolean {
     return this.isTypePutBackOn({ name: this.name })
   }
 
-  get notationRules (): Array<(v) => boolean | string> {
-    if (this.enableValidation) {
```

Comment:
I moved the enableValidation check to the template. I think it's a bit cleaner this way and it saves a tiny bit of processing because this (cached) getter is only evaluated once (doesn't need to be responsive), while the component was already responsive to changed rule.

Ask if you have any questions.